pkg/controllers/sync/controller.go

```diff
@@ -571,11 +571,12 @@ func (s *SyncController) syncToClusters(fedResource FederatedResource, collision
 		// Versioning of federated resources is an optimization to
 		// avoid unnecessary updates, and failure to record version
 		// information does not indicate a failure of propagation.
-		runtime.HandleError(err)
+		keyedLogger.WithValues("error-info", err).V(3).Info("Failed to record version information")
```

Comment:
```suggestion
		keyedLogger.Error(err, "Failed to record version information")
```
Although this error should not prevent sync from succeeding, we should still log the error for visibility.